pkg/api/controllers/fileshare.go

```diff
@@ -0,0 +1,288 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+
+	log "github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/api/util"
+	c "github.com/opensds/opensds/pkg/context"
+	"github.com/opensds/opensds/pkg/controller/client"
+	"github.com/opensds/opensds/pkg/db"
+	"github.com/opensds/opensds/pkg/model"
+)
+
+func NewFileSharePortal() *FileSharePortal {
+	return &FileSharePortal{
+		CtrClient: client.NewClient(),
+	}
+}
+
+type FileSharePortal struct {
+	BasePortal
+
+	CtrClient client.Client
+}
+
+func (f *FileSharePortal) CreateFileShare() {
+	var fileshare = model.FileShareSpec{
+		BaseModel: &model.BaseModel{},
+	}
+
+	// Unmarshal the request body
+	if err := json.NewDecoder(f.Ctx.Request.Body).Decode(&fileshare); err != nil {
+		reason := fmt.Sprintf("Parse fileshare request body failed: %s", err.Error())
+		f.Ctx.Output.SetStatus(model.ErrorBadRequest)
+		f.Ctx.Output.Body(model.ErrorBadRequestStatus(reason))
+		log.Error(reason)
+		return
+	}
+	result, err := util.CreateFileShareDBEntry(c.GetContext(f.Ctx), &fileshare)
+	if err != nil {
+		reason := fmt.Sprintf("Create fileshare failed: %s", err.Error())
+		f.Ctx.Output.SetStatus(model.ErrorBadRequest)
```

Comment:
You can use `f.ErrorHandle(model.ErrorBadRequest, reason)` here.

Reply:
done. Thanks